compose/project.py

```diff
@@ -115,6 +117,14 @@ def from_dicts(cls, name, service_dicts, client, use_networking=False, network_d
                     net=net,
                     volumes_from=volumes_from,
                     **service_dict))
+        if config_data.volumes:
+            for vol_name, data in config_data.volumes.items():
+                project.volumes.append(
+                    Volume(
+                        client=client, project=name, name=vol_name,
+                        driver=data.get('driver'), driver_opts=data.get('driver_opts')
+                    )
```

Comment:
Oh ya, just looking at `service.py` it seems the user would actually have to specify a value with the prefix included, is that right?

Another PR sounds fine to me.  

By the time this is in a release, we should have some way to use externally created volumes. #2502 is the issue for it.